Rare earth processing chain

WideMann opened the pull request:
This PR is about complete rare earth processing. From ore processing, through element separation to purification of rare earth elements

trainvoi's list of things that must be done before merging this PR:

- [ ] move all solutions to SecondDegreeMaterial (example: YCl3.H2O counts as a SecondDegreeMaterial, since YCl3 and H2O exist as a FirstDegreeMaterial)
- [ ] register different colors for new materials registered in UnknownCompositionMaterials.groovy
- [ ] prune materials that were registered but never used in this PR

WideMann committed:
Registering materials pt. 1

Materials for general processing line and monazite processing

WideMann committed:
missing resin

WideMann committed:
More changes

WideMann committed:
Bastnasite processing pt.1

No recipes yet

WideMann committed:
Registering Bastnasite pt.2

WideMann committed:
Merge branch 'SymmetricDevs:master-ceu' into stuffadditions

bruberu commented on groovy/material/UnknownCompositionMaterials.groovy:
```diff
@@ -2076,5 +2076,35 @@ public class UnknownCompositionMaterials {
+                .color()
+                .build();
+
+        RoastedBastnasiteLeech = new Material.Builder(4522, SuSyUtility.susyId('roasted_bastnasite_leech'))
```
Technically, this should be "leach" rather than "leech," as the latter refers to blood-sucking worms.

WideMann committed:
Bastnasite processing pt.2

2nd processing material registration

WideMann committed:
Bastnasite group separation

WideMann commented:
Can I use acronyms in naming the materials, or do we "Waxed Lightly Weathered Cut Copper Stairs" them

WideMann committed:
Bastnasite LREE processing, localization pt.1

WideMann committed:
More work

i don't know how to name this pr

WideMann committed:
reidification + spelling mistaek

WideMann committed:
Merge branch 'master-ceu' into stuffadditions

WideMann committed:
A lot of work

Processing resources registration,
Xenotime and moznite localization
some textures for resins
some element processing

WideMann committed:
Remove Krita autosave file

Uh oh, busted

WideMann committed:
Lines done, localization and number to-do

somehting not worky, also oved groovy file to fblock

WideMann committed:
Merge branch 'master-ceu' into stuffadditions

WideMann committed:
Processing ratios

only element specific ratios are left, and some localization

WideMann committed:
Merge remote-tracking branch 'origin/stuffadditions' into stuffadditions

WideMann committed:
Merge branch 'master-ceu' into stuffadditions

WideMann committed:
Pretty sure i'm done

disregard lack of waste recycling

bruberu commented on resources/gregtech/textures/items/metaitems/beads.chlorinated_strong_basic_anion_exchange.png:
Was anything going to happen with this texture?

WideMann replied:
It is a remnant from old idea, I will delete it

WideMann committed:
Removal of old

WideMann committed:
Merge remote-tracking branch 'origin/stuffadditions' into stuffadditions

WideMann committed:
LREE fixes and waste recyling pt.1

WideMann committed:
Final Steps

Added MREE & HREE waste processing,
Sodium chromate
Localization
Pr & Nd Hydroxides

Rick8484 commented on groovy/postInit/chemistry/inorganic_chemistry/RareEarthProcessing.groovy:
```diff
@@ -7,15 +8,89 @@ import static gregtech.api.unification.ore.OrePrefix.dye;
+
+GRAVITY_SEPARATOR.recipeBuilder()
+    .inputs(ore('dustBastnasite'))
+    .outputs(metaitem('dustSiftedBastnasite'))
```
byproducts are missing
Limestone tailing (30%) dust and Barite dust (5%)

Rick8484 commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/PraseodymiumChain.groovy:
```diff
@@ -4,20 +4,32 @@ import static gregtech.api.unification.material.Materials.*;
+
+ROASTER.recipeBuilder()
+        .inputs(ore('dustPraseodymiumHydroxide') * 18)
+        .outputs(metaitem('dustPraseodymiumOxide') * 3)
```
2Pr(OH)4 -> 2PrO2 + 4H2O 
meaning its should be 6 PrO2 dust not 3
or just use simplified reaction 
Pr(OH)4 -> PrO2 + 2H2O 
meaning 9 Pr(OH)4 dust-> 3 PrO2 dust + 2 H2O gas

WideMann committed:
Rick FIx

WideMann committed:
Merge branch 'master-ceu' into stuffadditions

bruberu commented on groovy/classes/ChangeFlags.groovy:
```diff
@@ -203,6 +203,8 @@ class ChangeFlags {
         Diatomite.setFormula("(SiO2)8(Fe2O3)(Al2O3)", true);
         Pollucite.setFormula("(Cs,Na)2Al2Si4O12(H2O)2", true);
         Pitchblende.setFormula("(?)UO2", true);
+        Bastnasite.setFormula("(La, Ce, Y)CO3F", true);
```
Remove spaces here.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/PraseodymiumChain.groovy:
```diff
@@ -0,0 +1,35 @@
+REACTION_FURNACE = recipemap('reaction_furnace')
+ROASTER = recipemap('roaster')
+
+ROASTER.recipeBuilder()
```
I do need a little bit of evidence that Pr(OH)4 would actually exist, given that other resources say it does not.

WideMann committed:
Bruberu Fix

WideMann committed:
Merge remote-tracking branch 'origin/stuffadditions' into stuffadditions

bruberu commented on groovy/material/FirstDegreeMaterials.groovy:
```diff
@@ -3661,10 +3665,326 @@ public class FirstDegreeMaterials {
+                .components(Cerium * 1, Nitrogen * 4, Oxygen * 12)
+                .colorAverage()
+                .build()
+                .setFormula("Ce(NO3)4", true)
```
You may want to separate out the setFormula call into its own line.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/actinides/UraniumChain.groovy:
```diff
@@ -257,4 +257,6 @@ REACTION_FURNACE.recipeBuilder()
\ No newline at end of file
+    .buildAndRegister()
+
+// From Monazite Processing
```
Is there something that's supposed to be here?

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/GadoliniumChain.groovy:
```diff
@@ -0,0 +1,45 @@
+MIXER.recipeBuilder()
+        .inputs(metaitem('beads.loaded_gadolinium_anion_exchange'))
+        .fluidInputs(fluid('distilled_water') * 1000)
+        .outputs(metaitem('dustGadoliniumHydroxide'))
```
The act of the disappearing beads

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+
+BR.recipeBuilder()
+        .inputs(ore('dustConcentratedBastnasite'))
+        .fluidInputs(fluid('sodium_hydroxide_solution') * 2000)
```
Needs to be 4 buckets

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .fluidInputs(fluid('sifted_monazite_leach') * 1000)
+        .fluidInputs(fluid('ammonia_solution') * 1000)
+        .outputs(metaitem('dustThoriumHydroxidePrecipitate') * 7)
+        .fluidOutputs(fluid('thfree_monazite_leach') * 1000)
```
Make sure to include something for processing the thorium hydroxide and uranyl hydroxide precipitates.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .notConsumable(metaitem('item_filter'))
+        .fluidInputs(fluid('xenotime_slurry') * 4000)
+        .outputs(metaitem('dustXenotimeResidue'))
+        .fluidOutputs(fluid('xenotime_phosphate_waste') * 4000)
```
Don't forget about the 8 moles of sodium that are in this from the soda ash when you make a recipe for this.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+
+SIFTER.recipeBuilder()
+        .inputs(ore('dustXenotimeTailing'))
+        .chancedOutput(metaitem('dustCassiterite') * 3, 500, 0)
```
These are probably all supposed to be nitrates, given that this was just dumped in nitric acid.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+// Bastnasite Concentrate Processing
+
+MIXER.recipeBuilder()
+        .inputs(ore('dustBastReHydroxides'))
```
This step and the next one can be combined (and also remove the bast_re_nitrates_solution)

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+MIXER.recipeBuilder()
+        .fluidInputs(fluid('hydrochloric_acid') * 50)
+        .fluidInputs(fluid('xeno_dehpa_extract') * 1000)
+        .fluidOutputs(fluid('rare_earth_extraction_mixture') * 1000)
```
What particularly makes this step require a full bucket of rare earth separation mixture as opposed to the 100L used elsewhere?

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .buildAndRegister()
+
+MIXER.recipeBuilder()
+        .notConsumable(fluid('sodium_hydroxide') * 432)
```
I mean, there are other molten NaOH recipes in SUSY, but I do want to see what the source on this one is, especially after 10 moles of NaOH were added earlier.

WideMann replied:
![image](https://github.com/user-attachments/assets/0015bd62-b54b-4ea5-bb60-11a527bce701)
As it was explained to me fused potassium hydroxide means molten potassium hydroxide. As we are using sodium compounds instead of potassium, this has been changed to molten sodium hydroxide.

bruberu replied:
In this case, it doesn't really seem helpful to include the previous step where the NaOH is already added, since it seems like you would just melt the NaOH dust anyways. This could allow you to save some material IDs, at least.

WideMann replied:
The previous step is to make RE hydroxides from RE nitrates, I don't think the steps can be made into one, as that fragment states that RE hydroxides were in a solution of molten hydroxide

bruberu replied:
Oh I do now see the 45 sodium nitrate dust at the end.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+LCR.recipeBuilder()
+        .fluidInputs(fluid('bast_eufree_mree_concentrate') * 1000)
+        .fluidInputs(fluid('sodium_amalgam') * 800)
+        .fluidOutputs(fluid('samarium_amalgam') * 800)
```
I would also like to see where this particularly came from, since I don't see much about this online.

WideMann replied:
![image](https://github.com/user-attachments/assets/865b7197-d2f5-4ae5-9428-1cc0dd59c3c2)
I might have did errors translating the book into recipes, but the main gist (using sodium amalgam to extract) is still here. Europium is reduced previously by zinc, so it's not taken into consideration during sodium amalgam extractions

bruberu replied:
Alright, although it doesn't quite seem that you've included the acetate here, which you may want to add (as it may otherwise not work).

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .fluidInputs(fluid('bast_lree_waste') * 1000)
+        .outputs(metaitem('dustPotassiumPermanganate') * 4)
+        .outputs(metaitem('dustManganeseIiNitrate') * 12)
+        //.outputs(metaitem('dustSodiumNitrate') * 45) Uncomment when crystallizer has 3 output slots
```
Have you made a PR to SusyCore for this?

WideMann replied:
Yes, merged by trainvoi

bruberu replied:
I think it should then be possible for you to just merge master-ceu back in and then uncomment these.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .outputs(metaitem('dustSodiumChromate') * 14)
+        .outputs(metaitem('dustSmallSodiumDodecylSulfate') * 13)
+        .chancedOutput(metaitem('dustSmallSodiumDodecylSulfate'), 5000, 0)
+        //.chancedOutput(metaitem('dustSalt') * 6, 8766, 0) Uncomment when crystallizer has 3 output slots
```
Oh, you'll need 4 output slots, actually

WideMann replied:
I'll multiply it by 2 to not get chancedOutputs.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/SamariumChain.groovy:
```diff
@@ -0,0 +1,28 @@
+BR.recipeBuilder()
+        .fluidInputs(fluid('samarium_amalgam') * 1000)
+        .fluidInputs(fluid('hydrochloric_acid') * 5000)
+        .outputs(metaitem('dustSamariumChloride') * 4)
```
Wikipedia backs up that this is soluble enough to be contained in a solution (92.4g/100mL), although I think mercury chloride wouldn't be included due to the common ion effect.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/YtterbiumChain.groovy:
```diff
@@ -0,0 +1,28 @@
+BR.recipeBuilder()
+        .fluidInputs(fluid('ytterbium_amalgam') * 1000)
+        .fluidInputs(fluid('hydrochloric_acid') * 5000)
+        .outputs(metaitem('dustYtterbiumChloride') * 4)
```
Again, given the very large amount of water, ytterbium chloride would probably manage to dissolve in it (17 g/100 mL).

bruberu commented on groovy/postInit/chemistry/organic_chemistry/Extractants.groovy:
```diff
@@ -1004,3 +1005,63 @@ MIXER.recipeBuilder()
+        .EUt(40)
+        .buildAndRegister()
+
+DISTILLERY.recipeBuilder()
```
It's late for me, but I'm fairly suspicious of this partial dimerization, given that I'm not quite sure what happens to the second phosphate.

bruberu replied:
OK, it appears that you should be able to just change the previous recipe to use 100L of 2-ethylhexanol to just directly make EHEHPA. (Not so sure about the catalyst bed, but it's fine for now.)

WideMann replied:
EHEHPA production changed completely

Crindigo commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/PraseodymiumChain.groovy:
```diff
@@ -0,0 +1,35 @@
+        .inputs(ore('dustPraseodymiumOxide') * 3)
+        .fluidInputs(fluid('hydrofluoric_acid') * 4000)
+        .outputs(metaitem('dustPraseodymiumFluoride') * 5)
+        .fluidOutputs(fluid('dense_steam') * 2000)
```
dense steam output should be 6000L
PrO2 + 4(HF)(H2O) -> PrF4 + 6(H2O)

WideMann committed:
Bruberu fix #3

I hope it's fine now

WideMann committed:
Missing texture fix

WideMann committed:
Separation MIxture ratio

WideMann committed:
HEHEHP/EHEHPA change

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .fluidInputs(fluid('thfree_monazite_leach') * 1000)
+        .fluidInputs(fluid('ammonia_solution') * 1000)
+        .outputs(metaitem('dustMonaReHydroxides'))
+        .fluidOutputs(fluid('refree_monazite_leach') * 1000)
```
This and the following recipe should be a phase separator.

bruberu commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/RareEarthProcessing.groovy:
```diff
@@ -0,0 +1,1046 @@
+        .EUt(30)
+        .buildAndRegister()
+
+CENTRIFUGE.recipeBuilder()
```
This should also be a phase separator, as well as the monazite and xenotime versions, although fortunately it's already in the right form.

WideMann committed:
EHEHPA corrected production

trainvoi committed:
Merge remote-tracking branch 'upstream/master-ceu' into pr/1227

trainvoi committed:
merge fix

trainvoi commented:
im getting method too large errors with this PR

we might have to split up FirstDegreeMaterials

trainvoi committed:
merge fix

bruberu reviewed (commented):
Forgot to add these

planetme commented on groovy/classes/ChangeFlags.groovy:
```diff
@@ -203,6 +203,8 @@ class ChangeFlags {
         Diatomite.setFormula("(SiO2)8(Fe2O3)(Al2O3)", true);
         Pollucite.setFormula("(Cs,Na)2Al2Si4O12(H2O)2", true);
         Pitchblende.setFormula("(?)UO2", true);
+        Bastnasite.setFormula("(La,Ce,Y)CO3F", true);
```
should include all components with composition >1%

planetme commented on groovy/material/SuSyMaterials.groovy:
```diff
@@ -1315,6 +1317,177 @@ class SuSyMaterials {
         public static Material WeaponsGradeUraniumDioxide;
         public static Material DepletedUraniumDioxide;
 
+        // Rare Earth Processing Materials
```
many of these materials needs to be placed in their correct categories

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3930,12 +3930,12 @@ mods.gregtech.fluid_solidifier.recipeBuilder()
 
 // Iron III Nitrate Solution
 BR.recipeBuilder()
```
please use 4 tab indents...

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3974,6 +3974,114 @@ MIXER.recipeBuilder()
+        .EUt(30)
+        .buildAndRegister()
+
+DISTILLERY.recipeBuilder()
```
needs to be batch not continuous

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3974,6 +3974,114 @@ MIXER.recipeBuilder()
+
+// Phosphorous Acid
+
+CSTR.recipeBuilder()
```
wrong division

planetme replied:
also needs more water, phosphorous acid is a solid, should be 3 hcl 3 h2o and 1 h3po3 in the solution

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3974,6 +3974,114 @@ MIXER.recipeBuilder()
+        .buildAndRegister()
+
+
+ROASTER.recipeBuilder()
```
can be made a smaller recipe

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3984,4 +4092,60 @@ BR.recipeBuilder()
+
+// EHEHPA Production
+
+BR.recipeBuilder()
```
where is the 2-ethylhexyl group coming from lol, also we have tscl already

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3984,4 +4092,60 @@ BR.recipeBuilder()
+BR.recipeBuilder()
+        .inputs(ore('dustPotassiumIodide') * 2)
+        .fluidInputs(fluid('2_ethylhexyl_tosylate') * 1000)
+        .fluidOutputs(fluid('potassium_tosylate') * 1000)
```
needs recycle

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3984,4 +4092,60 @@ BR.recipeBuilder()
+        .EUt(120)
+        .buildAndRegister()
+
+CSTR.recipeBuilder()
```
wrong subdivision of recipe

planetme replied:
wrong subdivision of recipe

planetme commented on groovy/postInit/chemistry/ChemistryOverhaul.groovy:
```diff
@@ -3984,4 +4092,60 @@ BR.recipeBuilder()
+        .notConsumable(fluid('acetic_acid') * 500)
+        .fluidInputs(fluid('water') * 1000)
+        .fluidInputs(fluid('ethylhexyl_phosphonic_acid_di_2_ethylhexyl_ester') * 1000)
+        .fluidOutputs(fluid('two_ethylhexanol') * 2000)
```
wrong amt

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/actinides/Thorium.groovy:
```diff
@@ -0,0 +1,13 @@
+import javax.sound.sampled.Mixer
```
why

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/actinides/Thorium.groovy:
```diff
@@ -0,0 +1,13 @@
+
+BR = recipemap('batch_reactor')
+
+BR.recipeBuilder()
```
this is not how you implement LLE

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/actinides/UraniumChain.groovy:
```diff
@@ -1,3 +1,5 @@
+import javax.sound.sampled.Mixer
```
why

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/actinides/UraniumChain.groovy:
```diff
@@ -230,4 +232,16 @@ FLUIDIZEDBR.recipeBuilder()
+
+// From Monazite Processing
+
+BR.recipeBuilder()
```
no...

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/CeriumChain.groovy:
```diff
@@ -0,0 +1,36 @@
+        .inputs(ore('dustCeriumIvNitrate') * 17)
+        .outputs(metaitem('dustCeriumIvOxide') * 3)
+        .fluidOutputs(fluid('nitrogen_dioxide') * 4000)
+        .fluidOutputs(fluid('oxygen') * 2000)
```
what..

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/CeriumChain.groovy:
```diff
@@ -0,0 +1,36 @@
+BR.recipeBuilder()
+        .inputs(ore('dustCeriumIvOxide') * 3)
+        .fluidInputs(fluid('hydrofluoric_acid') * 4000)
+        .outputs(metaitem('dustCeriumFluoride') * 5)
```
should be done with pure fluorine

planetme commented on groovy/postInit/chemistry/inorganic_chemistry/elements/f_block/lanthanides/CeriumChain.groovy:
```diff
@@ -0,0 +1,36 @@
+        .outputs(metaitem('dustFluorite') * 6)
+        .duration(200)
+        .EUt(960)
+        .buildAndRegister()
```
needs high purity

planetme committed:
cleanup, bast beneficiation changes

planetme committed:
reorganization

planetme reviewed (commented):
notes for changes pt 1

planetme committed:
Merge branch 'master-ceu' into pr/1227

planetme committed:
this will break things

planetme committed:
finish bast

planetme committed:
mona line

bruberu committed:
Merge branch 'master-ceu' into stuffadditions

planetme committed:
xenotime chain

planetme committed:
some chain changes

trainvoi committed:
formatting cleanup

trainvoi committed:
Merge remote-tracking branch 'upstream/master-ceu' into pr/1227

planetme committed:
ready for review

WideMann commented on groovy/postInit/chemistry/inorganic_chemistry/elements/d_block/group3/YttriumChain.groovy:
```diff
@@ -0,0 +1,41 @@
+DISTILLERY.recipeBuilder()
+    .fluidInputs(fluid('yttrium_chloride_solution') * 6000)
+    .outputs(metaitem('dustYttriumChloride') * 4)
+    .fluidOutputs(fluid('water') * 1000)
```
Should be 6000kL of water

h3tR replied:
> Should be 6000kL of water

6000 Buckets of water???????

planetme committed:
skull

bruberu committed:
fix: second degree materials error

bruberu reviewed (approved):
It is time.
